lnrpc/routerrpc/router_backend.go

```diff
@@ -44,6 +45,11 @@ type RouterBackend struct {
 	// SelfNode is the vertex of the node sending the payment.
 	SelfNode route.Vertex
 
+	// FetchChannelInfo is a closure that we'll use the fetch the latest
+	// routing state for a particular channel identified by its channel ID.
+	FetchChannelInfo func(chanID uint64) (*models.ChannelEdgeInfo,
```

Comment:
Although we are currently only interested in a bool return value, we might still return the full chaninfo as a prevision for future use case, hmm I am not sure maybe we just simplify the function signature for now and only return a bool (also then maybe changing the name of the function)?

Reply:
We could also repurpose some of the other endpoints exposed here already, like `FetchChannelCapacity` or `FetchChannelEndpoints`, which would return `EdgeNotFound`

Reply:
To keep the footprint of this PR small I agree to repurpose `FetchChannelEndpoints` to check if we deal with a public route hint. I've removed the field `FetchChannelInfo` from the router backend.